test/unittest/tests/test_session_config.cpp

```diff
@@ -0,0 +1,66 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "wt_internal.h"
+#include "wrappers/mock_session.h"
+
+/*
+ * Check a basic configuration that sets and clears a flag.
+ */
+#define FLAG_TEST(config_param, flag)                                                    \
+    TEST_CASE(config_param, "[session_config]")                                          \
+    {                                                                                    \
+        std::shared_ptr<MockSession> session_mock = MockSession::buildTestMockSession(); \
+        WT_SESSION_IMPL *session = session_mock->getWtSessionImpl();                     \
+        session->flags = 0;                                                              \
+        REQUIRE(__ut_session_config_int(session, config_param "=true") == 0);            \
```

Comment:
We can also test for invalid arguments here. The [csuite config test](https://github.com/wiredtiger/wiredtiger/blob/develop/test/csuite/config/main.c) has some examples we can reference.

Reply:
I think at that point we're testing __wt_config_getones and not the function itself which is why I've avoided this.